pkg/token/session.go

```diff
@@ -0,0 +1,35 @@
+package token
+
+import (
+	"github.com/google/uuid"
+)
+
+type SessionToken struct {
+	id     uuid.UUID
+	pubKey []byte
+}
+
+func CreateSessionToken(id, pub []byte) (*SessionToken, error) {
+	var tokenID uuid.UUID
+
+	err := tokenID.UnmarshalBinary(id)
+	if err != nil {
+		return nil, err
+	}
+
+	key := make([]byte, len(pub))
+	copy(key[:], pub)
+
+	return &SessionToken{
+		id:     tokenID,
+		pubKey: key,
+	}, nil
+}
+
+func (s SessionToken) SessionKey() []byte {
+	return s.pubKey
+}
+
+func (s SessionToken) ID() []byte {
+	return s.id[:]
```

Comment:
I suggest `return s.id.Bytes()`

Reply:
`uuid.UUID` doesn't have `Bytes()` method, only `MarshalBinary()` that returns `([]byte, error)`

Reply:
Let's use `MarshalBinary` then, it never returns an error.

Reply:
Done